### PR TITLE
Fix python implementation pickling

### PR DESCRIPTION
**Pull request by apmorton**
This PR is split into two commits for easier review since git doesn't correctly show the rename operation on `core.py` 😞 .

The first moved the pure python implementation to `_frozendictpy` so that `core.py` can become a compatibility wrapper, but makes no other changes. This allows existing pickles that mention `frozendict.core.frozendict` to correctly depickle as the native implementation in the future on python 3.11.

The second implements the fix for #87.

### CODE REVIEW

**Comment by Marco-Sulla** [comment]
Excuse me, but why you have renamed also `core.py`? Fixing #87 was not sufficient?

PS: I added a build and test for the pure py implementation in the pipeline, so I'm going to remove the pure py tests from the C extension tests.

Wait for it.

**Comment by apmorton** [comment]
> Excuse me, but why you have renamed also core.py?

I explained this in the PR description.

Existing files that have been pickled on python 3.11 mention `frozendict.core.frozendict`.
This means that if/when 3.11 gets a native extension these files will still depickle to the pure python implementation and you can end up with mixed native/python instances in a single process.

Moving the python implementation and setting up `frozendict.core.frozendict` to be an alias for `frozendict.frozendict` resolves this issue.

**Comment by Marco-Sulla** [comment]
> This means that if/when 3.11 gets a native extension these files will still depickle to the pure python implementation and you can end up with mixed native/python instances in a single process.

Ok, but what if I pickle `frozendict._frozendictpy.frozendict` and then I switch to the C extension?

At this point, is it not much simpler to remove `core.py`?

**Comment by apmorton** [comment]
> Ok, but what if I pickle `frozendict._frozendictpy.frozendict` and then I switch to the C extension?

If you pickle `frozendict._frozendictpy.frozendict` it will appear in the pickle file as `frozendict.frozendict` because it has `__module__` set to `frozendict` and depickle correctly in the future using the C extension.

File: src/frozendict/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
     c_ext = True
     del _frozendict
 except ImportError:
-    from .core import *
+    from ._frozendictpy import *
```
Comment (Marco-Sulla) [inline src/frozendict/__init__.py:10]: Emh. I don't like the name at all... can you change it with `heart`?

File: src/frozendict/core.py [hunk 2]
```diff
@@ -1,583 +1,4 @@
...
-        else:
-            res = self & other
-        
-        return not res
-
-__all__ = (frozendict.__name__, )
+# this provides compatibility for older pickles
+# created on a python-only implementation that
+# specifically mention frozendict.core.frozendict
+from frozendict import frozendict
```
Comment (Marco-Sulla) [inline src/frozendict/core.py:4]: Just a curiosity: how this works?

File: test/common.py [hunk 3]
```diff
@@ -112,6 +112,8 @@ def test_pickle(self, fd, protocol):
         dump = pickle.dumps(fd, protocol=protocol)
         assert dump
         assert pickle.loads(dump) == fd
+        assert b'core' not in dump
```
Comment (Marco-Sulla) [inline test/common.py:115]: These are good as one-time test, IMHO are not good as unit test

File: test/frozendict_only.py [hunk 4]
```diff
@@ -41,3 +42,11 @@ def test_init(self, fd):
     def test_del_empty(self):
         fd = self.FrozendictClass({1: 2})
         assert fd.delete(1) is self.FrozendictClass()
+
+    def test_load_core_pickle(self):
```
Comment (Marco-Sulla) [inline test/frozendict_only.py:46]: The same before: good for one shot tests, not for unit tests

**Review by Marco-Sulla — CHANGES_REQUESTED** [review]

File: test/frozendict_only.py (hunk 4 above)
Comment (apmorton) [reply test/frozendict_only.py:46]: Similar to above, I think it is important to have a check that my `src/frozendict/core.py` compatibility code actually works.  I can improve this test if you want, as I do agree hard coding some pickle bytes is not ideal.

File: src/frozendict/__init__.py (hunk 1 above)
Comment (Marco-Sulla) [reply src/frozendict/__init__.py:10]: Ok, what about `_frozendict_py`?

File: test/frozendict_only.py (hunk 4 above)
Comment (Marco-Sulla) [reply test/frozendict_only.py:46]: > I can improve this test if you want, as I do agree hard coding some pickle bytes is not ideal.

Well, let's see :)

File: src/frozendict/core.py (hunk 2 above)
Comment (Marco-Sulla) [reply src/frozendict/core.py:4]: How do you discovered it?

**Review by Marco-Sulla — APPROVED** [review]

**Pull request merged**